client/signup/main.jsx

```diff
@@ -183,13 +188,13 @@ const Signup = React.createClass( {
 	componentDidMount() {
 		debug( 'Signup component mounted' );
 		SignupProgressStore.on( 'change', this.loadProgressFromStore );
-		SignupProgressStore.on( 'change', this.loadDependenciesFromStore );
+		this.sigunpListener = SignupDependencyStore.reduxStore.subscribe( this.loadDependenciesFromStore );
```

Comment:
That was an awesome catch, updated to be more descriptive - 6daec3e2110d9ed77a3aeaca1fe02e53350fca2c